lib/Statocles/App/Perldoc.pm

```diff
@@ -212,12 +214,38 @@ sub pages {
                 crumbtrail => \@crumbtrail,
             },
         );
-
+        # unable to highlight source as source.html.ep uses <%== %> to escape html.
+        # $self->_highlight_page( $pages[-1], 'pre' );
     }
 
     return @pages;
 }
 
+sub _highlight_page {
+  my ( $self, $page, $sel ) = @_;
+  # highlight only if site-wide highlighting is available
+  return unless my $hl = $page->site->plugins->{highlight};
+  # only highlight the page content
+  my $dom = Mojo::DOM->new($page->content);
+  my $codes = $dom->find($sel);
+  if ($codes->first) {
+    for my $node ($codes->each) {
+      my $parent = $node->tag eq 'code' ? $node->parent : $node;
+      my $highlighted =
+        Mojo::DOM->new($hl->highlight({page => $page}, Perl => $node->text))
```

Comment:
The HTML in [`$output`](https://github.com/preaction/Statocles/blob/master/lib/Statocles/Plugin/Highlight.pm#L183-L185). An alternative would be a regex 
```perl
s{<pre>(<code.*</code>)</pre>}{$1};
```

Reply:
Could you replace $parent? I just found [Mojo::DOM's replace() method](http://mojolicious.org/perldoc/Mojo/DOM#replace). Then it could just be:

```
$parent->replace( $hl->highlight( ... ) );
```

Reply:
Ah, I see. This should work - although it still calls parse, the find is saved.